### DISCUSSION TITLE
Concatenation failed

### DISCUSSION THREAD

**Discussion by MargaridaS** [SciTools/iris#5203] [Q&A] [answered]
Dear SciTools iris team,

I contact you because I've been doing some tests to concatenate 2 files without success and it would be nice if you could give me some hint. It seems that the metadata is the same but there is no way to concatenate them. The code I use to concatenate is the next one:
```
>>> import iris
>>> from iris.experimental.equalise_cubes import equalise_attributes
>>> ds = iris.load(['tasmax_201403.nc', 'tasmax_201404.nc'])
>>> equalise_attributes(ds)
>>> print(ds.concatenate_cube())
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/gpfs/projects/bsc32/software/suselinux/11/software/Iris/2.4.0-foss-2019b-Python-3.7.4/lib/python3.7/site-packages/scitools_iris-2.4.0-py3.7.egg/iris/cube.py", line 528, in concatenate_cube
    raise iris.exceptions.ConcatenateError(msgs)
iris.exceptions.ConcatenateError: failed to concatenate into a single cube.
  An unexpected problem prevented concatenation.
  Expected only a single cube, found 2.
```
I attach the files:
[tasmax.gz](https://github.com/SciTools/iris/files/11039574/tasmax.gz)
Thanks a lot in advance.
Best regards,
MargaridaS

**Comment by lbdreyer** [comment, marked as answer]
It looks like the problem may be due to the time coordinates.
`tasmax_201403.nc` produces a time coordinate:
```
DimCoord :  time / (hours since 1850-1-1 00:00:00, standard calendar)
    points: [2014-03-31 23:00:00]
    bounds: [[2014-04-16 13:30:00, 2014-03-16 08:30:00]]
    ...
```
 and `tasmax_201404.nc` produces a time coordinate:
```
DimCoord :  time / (hours since 1850-1-1 00:00:00, standard calendar)
    points: [2014-04-30 23:00:00]
    bounds: [[2014-04-01 00:00:00, 2014-04-30 23:00:00]]
    ...
```

So the time bounds are overlapping and one has the bounds increasing and the other has it decreasing.

Iris requires dimension coords to be monotonic. Due to the bounds of the input cubes overlapping and increasing vs decreasing, Iris can't create a dimension coordinate for time for the concatenate cube and so it fails.

To confirm this was the only problem preventing concatenation, I tried removing the bounds (i.e. `cube.coord('time').bounds=None` for each cube) and after doing so it would concatenate. So I think you just need to correct the bounds and then it should work

**Comment by MargaridaS** [comment]
Hi @lbdreyer 

Thank you very much for the information. Removing time_bnds and re-adding them solved the issue and concatenation worked properly. Thanks a lot!

Best regards,
MargaridaS
